### ISSUE TITLE
Sharing Patient Data With Primary Care Provider 

### ISSUE THREAD

**Issue by MihirJoe** [carekit-apple/CareKit#450]
I've been looking for the best way to share patient symptoms and medication adherence with a primary care provider for some time now and I've come to the conclusion that using a secure database––which the provider also has access to––is the route I should take. Since I'm new to CoreData and databases in general, I was wondering if this was the best way of going about doing things. Are there any specific platforms I should be using that are HIPAA compliant or could I utilize the Health app and share the information that way. The clinic I'm working with doesn't have that great of an EHR system. Are there any resources I can take a look at to get started with using databases with CareKit?

**Comment by MihirJoe** [comment]
Is the CareKit Remote Synchronization functionality related to this?

**Comment by marco-theraforge** [comment]
Hi Mihir,

we are working on a secure cloud database that may be of help to you. If you'd like, we can have a chat to share thoughts and see if we can be of assistance in your current endeavor.
Best,
Marco
marco@hippocratestech.com

**Comment by cbaker6** [comment]
@MihirJoe you are correct, `OCKRemoteSynchronizable` is designed to solve your problem. Any vendor can leverage `OCKRemoteSynchronizable` to synchronize data to a remote. My research lab has created an open source solution that you can use called [ParseCareKit](https://github.com/netreconlab/ParseCareKit) that leverages a HIPAA compliant parse-server, called [parse-hipaa](https://github.com/netreconlab/parse-hipaa).

In short, leveraging ParseCareKit will let you synchronize your CareKit data to the cloud with a few lines of code (I believe this is the attempt of any framework that uses `OCKRemoteSynchronizable`). If you used the OCKSample app to get started with your app, you should look at [CareKitSample-ParseCareKit](https://github.com/netreconlab/CareKitSample-ParseCareKit) which will help you get started with testing your current app with ParseCareKit and parse-hipaa (via docker). If you like it, feel free to keep using it (My team uses it in our research apps, so it will remain up-to-date). If it doesn't seem like the best solution for you, you can "docker-compose down" and your data is destroyed removing the framework from your app is easy as removing the cocoapod.

If you have questions, feel free to respond here or on ParseCareKit or CareKitSample-ParseCareKit.

**Comment by MihirJoe** [comment]
Thank you for the responses! I will definitely check them out. I am wondering whether or not simply emailing a screenshot to the care provider is a recommended way of sharing information.

**Comment by marco-theraforge** [comment]
> Thank you for the responses! I will definitely check them out. I am wondering whether or not simply emailing a screenshot to the care provider is a recommended way of sharing information.

I think that any loosely controlled sharing of patient data can become problematic in light of tighter and tighter privacy regulations.
The best approach is to use a database with proper access controls and encryption at rest, which can be accessed on demand exclusively by patient-approved individuals or groups. The system can be audited to guarantee proper security and information management. And patient data can be completely deleted when the right to be forgotten is exercised.
There is obviously no magic technology to deal with all possible scenarios, but by leveraging the state of the art it is certainly possible to make sure that in general patients remain in control of their medical data, even when it gets replicated, and don't inadvertently or unknowingly lose control over it...

My 2 cents,

Marco
